homeassistant/components/kostal_plenticore/select.py

```diff
@@ -0,0 +1,129 @@
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, SELECT_SETTINGS_DATA
+from .helper import Plenticore, SelectDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Add kostal plenticore Select widget."""
+    plenticore: Plenticore = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        PlenticoreDataSelect(
+            hass=hass,
+            plenticore=plenticore,
+            entry_id=entry.entry_id,
+            platform_name=entry.title,
+            device_class="kostal_plenticore__battery",
+            module_id=select.module_id,
+            data_id=select.data_id,
+            name=select.name,
+            current_option="None",
+            options=select.options,
+            is_on=select.is_on,
+            device_info=plenticore.device_info,
+            unique_id=f"{entry.entry_id}_{select.module_id}",
+        )
+        for select in SELECT_SETTINGS_DATA
+    )
+
+
+class PlenticoreDataSelect(CoordinatorEntity, SelectEntity, ABC):
+    """Representation of a Plenticore Switch."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        plenticore: Plenticore,
+        entry_id: str,
+        platform_name: str,
+        device_class: str | None,
+        module_id: str,
+        data_id: str,
+        name: str,
+        current_option: str | None,
+        options: list[str],
+        is_on: str,
+        device_info: DeviceInfo,
+        unique_id: str,
+    ) -> None:
+        """Create a new switch Entity for Plenticore process data."""
+        super().__init__(
+            coordinator=SelectDataUpdateCoordinator(
+                hass,
+                _LOGGER,
+                "Select Data",
+                timedelta(seconds=30),
+                plenticore,
+            )
+        )
+        self.plenticore = plenticore
+        self.entry_id = entry_id
+        self.platform_name = platform_name
+        self._attr_device_class = device_class
+        self.module_id = module_id
+        self.data_id = data_id
+        self._attr_options = options
+        self.all_options = options
+        self._attr_current_option = current_option
+        self._is_on = is_on
+        self._device_info = device_info
+        self._attr_name = name or DEVICE_DEFAULT_NAME
+        self._attr_unique_id = unique_id
+
+    @property
+    def available(self) -> bool:
+        """Return if entity is available."""
+        is_available = (
+            super().available
+            and self.coordinator.data is not None
+            and self.module_id in self.coordinator.data
+            and self.data_id in self.coordinator.data[self.module_id]
+        )
+
+        if is_available:
+            self._attr_current_option = self.coordinator.data[self.module_id][
```

Comment:
Entity state properties shouldn't have side effects. We shouldn't change another attribute in the available property. Use a separate callback to update entity state if needed.

Reply:
Fixed in https://github.com/home-assistant/core/pull/59998